Src/FluentAssertions/Execution/ContextDataItems.cs

```diff
@@ -42,10 +42,13 @@ public void Add(ContextDataItems contextDataItems)
 
         public void Add(DataItem item)
         {
-            DataItem existingItem = items.SingleOrDefault(i => i.Key == item.Key);
-            if (existingItem != null)
+            for (int i = 0; i < items.Count; i++)
```

Comment:
Why do you think that `for` is more efficient than `SingleOrDefault`?
I agree that "remove" and "add" is not that good, but `SingleOrDefault` is implemented in efficient way for all containers and even more better to read that the new for loop.

Reply:
I'm using the index inside the loop body. `SingleOrDefault` doesn't return the index.

Reply:
I can use `FindIndex` for this.